### ISSUE TITLE
Mint 19 Cinnamon: boots very slow

### ISSUE THREAD

**Issue by jeroenst** [linuxmint/linuxmint#33]
It takes about a minute to boot mint 19 cinnamon.

In dmesg I see te following delay:  

[    4.383680] Btrfs loaded, crc32c=crc32c-intel
[   36.972832] EXT4-fs (sda2): mounted filesystem with ordered data mode. Opts: (null)

Looks like mounting the EXT-4 fs takes time for an unknown reason

**Comment by jeroenst** [comment]
Seems that this issue is related to https://github.com/linuxmint/linuxmint/issues/34 because when I removed the /usr/share/initramfs-tools/scripts/local-premount/resume and runned the update-initramfs -ukall the system boots as fast as expected.

**Comment by jeroenst** [comment]
After creating a swapdevice and changing the uuid in /etc/initramfs-tools/conf.d/resume the system boots normal with the resume script in plase.

Also the errors about incorrect uuid for swapspace when running update-initramfs -ukall are gone.

**Comment by xenopeek** [comment]
See the Linux Mint 19 release notes. Assuming you have home directory encryption, there is a known issue with swap that you have to manually correct post-install as explained in the release notes. This known error may possibly not have been in the release notes at the date you reported above.

**Comment by jeroenst** [comment]
Should be fixed imo to get mint user friendly..

**Comment by BusinessKid** [comment]
This issue has resurfaced as a result of more recent updates. When I installed, I didn't have it. It came for me near the end of 2018, but update/upgrades have been sporadic as this is for me a spare distro. Swap existed (/dev/sda2), no encryption exists, no btrfs  exists, and I don't want it. This LQ thread discusses it
[https://www.linuxquestions.org/questions/linux-mint-84/irritating-delay-on-booting-mint-19-0-tara-4175648564/](url). The systemd-analyze commands showed nothing.  From dmesg:
[    6.391556] xor: automatically using best checksumming function   avx       
**[    6.409913] Btrfs loaded, crc32c=crc32c-intel
[   39.116244] EXT4-fs (sda3): mounted filesystem with ordered data mode. Opts: (null)**
[   39.277756] ip_tables: (C) 2000-2006 Netfilter Core Team
[   39.287204] systemd[1]: systemd 237 running in system mode. (+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD -IDN2 +IDN -PCRE2 default-hierarchy=hybrid)
[   39.308233] systemd[1]: Detected architecture x86-64.
[   39.320349] systemd[1]: Set hostname to <RoseViolet>.

The business of moving the 'resume' script, updating the initrd, didn't solve it either, nor did replacing that script. If I could give a middle finger to btrfs, that would work around it, but it's your bug, and it should be back open. 
My box is a 6 year old Samsung NP 350 i3/6G ram, 250G SSD, HD4000 Graphics on legacy boot. Nothing fancy or complicated, on legacy boot.

**Comment by iHad169** [comment]
xenopeek talk this:
[https://www.linuxmint.com/rel_tara_cinnamon.php](Linux Mint 19 release notes)
[https://forums.linuxmint.com/viewtopic.php?f=90&t=274157]([FIX] no swap on fresh LM19 install with home directory encryption)

xenopeek. Please write the LINK.
People do not know you talk about.

**Comment by leiwu123** [comment]
linux 19 mate takes forever to boot after fresh installation.  I wonder what the isssue is. Im trying this out after discovering so many bugs with ubuntu. So disappointed.

**Comment by cpitclaudel** [comment]
For me the problem was that the contents of /etc/initramfs-tools/conf.d/resume did not match the ID of my swap partition, as reported by `blkid` (look for `/dev/sdaX … TYPE="swap"`.  Correcting the discrepancy and running `sudo update-initramfs -c -k 'all'` was enough to fix the issue.

**Comment by leiwu123** [comment]
my linux mint works fabulously now after a reinstall when I chose my main SSD drive to install the bootdriver

**Comment by BusinessKid** [comment]
Whoever closed this bug was premature or hopeful.
1. I still have the bug. I tried @cpitclaudel 's trick about the resume script, & rebuilding the initrd and it achieved nothing.
2. Just to make it go away, I removed the btrfs module from /lib/modules, blacklisted it in /etc/modprobe.d/blacklist.conf, and rebuilt the initrd again. It still managed to load the initrd. It must have a initrd tree somewhere, and I didn't clean that out. I'm still on kernel 4.15.0-20. Should I have updated that, and how?
